File: day_06.R
data <- as.numeric(read.table("data/input_06.txt", header = FALSE, sep = ","))
counts <- as.numeric(table(factor(data, levels = 0:8)))
for (i in 1:256){
  old <- counts[1]
  counts <-  c(counts[-1],counts[1])
  counts[7] <- counts[7]+old
}

format(sum(counts), scientific = FALSE)

File: day_06.py
from collections import Counter
with open("data/input_06.txt") as data:
    lanternfish = data.readline().strip().split(",")
    lanternfish = list(map(int, lanternfish))

counts = [Counter(lanternfish)[i] for i in range(9)]

for i in range(256): # or 80
  old = counts[0]
  counts =  counts[1:] + [counts[0]]
  counts[6] = counts[6]+old

sum(counts)
